src/event-handlers/impl/UserEventHandler.js

```diff
@@ -0,0 +1,61 @@
+import BaseEventHandler from './../base/BaseEventHandler';
+import APP_EVENT from './../../events/base/AppEvent';
+import {
+  UserService
+} from './../../services';
+
+
+class UserEventHandler extends BaseEventHandler {
+
+  constructor() {
+    super();
+  }
+
+}
+
+const userEventHandler = new UserEventHandler();
+const userService = new UserService();
+
+userEventHandler.register(APP_EVENT.USER_CREATED, async (event) => {
+
+  const {
+    username,
+    status,
+    pubKey,
+    tenantId,
+    email,
+    attributes,
+    role
+  } = event.getEventPayload();
+
+  const createdUserProfile = await userService.createUserProfile({
+    username,
+    status,
+    signUpPubKey: pubKey,
+    tenant: tenantId,
+    email,
+    attributes,
+    roles: role ? [{
+      role: role,
+      researchGroupExternalId: tenantId
```

Comment:
Let's replace `role` in the `UserCreatedEvent` with `roles` array and do the same in the `CreateUserCmd`. We need to allow the UI to pass the `role` and `teamId` while registering a new user. This flow will be used in the Vedai while registering a bootcamp